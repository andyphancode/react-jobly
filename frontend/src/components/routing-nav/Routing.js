import React from 'react';
import { Route, Routes } from "react-router-dom";
import './Routing.css'
import Homepage from '../home/Homepage';
import CompanyList from '../companies/CompanyList';
import CompanyDetails from '../companies/CompanyDetails';
import JobList from '../jobs/JobList';
import LoginForm from '../auth/LoginForm';
import SignupForm from '../auth/SignupForm';
import ProfileForm from '../profile/ProfileForm';


function Routing({ login, signup }) {
  return (
      <Routes>
        <Route path="/" element={<Homepage/>} />
        <Route path="/companies" element={<CompanyList/>} />
        <Route path="/companies/:handle" element={<CompanyDetails/>} />
        <Route path="/jobs" element={<JobList/>} />
        <Route path="/login" element={<LoginForm login={login}/>} />
        <Route path="/signup" element={<SignupForm signup={signup}/>} />
        <Route path="/profile" element={<ProfileForm/>} />
      </Routes>        
  );
};

export default Routing;
