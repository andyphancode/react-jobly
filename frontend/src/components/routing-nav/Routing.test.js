import React from "react";
import { render } from "@testing-library/react";
import Routing from "./Routing";
import { MemoryRouter } from "react-router-dom";

it("renders without crashing", function () {
    render(
        <MemoryRouter>
            <Routing/>
        </MemoryRouter>
    )
});

it("renders without crashing", function () {
    const { asFragment } = render(
        <MemoryRouter>
            <Routing/>
        </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
}); 