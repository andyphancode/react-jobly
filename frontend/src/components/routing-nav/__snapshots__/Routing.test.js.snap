// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Welcome to our website!
    </h1>
  </div>
</DocumentFragment>
`;
